lib/tasks/db.rake

```diff
@@ -0,0 +1,16 @@
+namespace :db do
+  desc "Add constraint to votes on topics. Vote >= 0"
+  task add_constraints: :environment do
+    ActiveRecord::Base.connection.execute <<-SQL
+      ALTER TABLE topics ADD CONSTRAINT positive_votes CHECK (votes >= 0);
```

Comment:
Seems like i can write a custom error rescue in the controller. I'll Add that now :)